docs/tutorial/fdfitting.rst

```diff
@@ -0,0 +1,343 @@
+parameters in this model. So for instance, the persistence length in this model, would be
+named `DNA/Lp`. Let's have a look at the model and its parameters::
+
+    >>> odijk_forward = pylake.odijk("DNA")
+    >>> odijk_forward
+
+    Model equation:
+
+    d(f) = DNA.Lc * (1 - (1/2)*sqrt(kT/(f*DNA.Lp)) + f/DNA.St)
+
+    Parameter defaults:
+
+    Name      Value  Unit      Fitted      Lower bound    Upper bound
+    ------  -------  --------  --------  -------------  -------------
+    DNA/Lp    40     [nm]      True                  0            inf
+    DNA/Lc    16     [micron]  True                  0            inf
+    DNA/St  1500     [pN]      True                  0            inf
+    kT         4.11  [pN*nm]   False                 0              8
+
+Note how most of the parameters listed have the model name prefixed. The only parameter
+that doesn't is a parameter that is not expected to be model specific (namely the
+parameter representing the Boltzmann constant times the temperature). We can also
+obtain these parameters as a list::
+
+    >>> print(odijk_forward.parameter_names)
+
+    ['DNA/Lp', 'DNA/Lc', 'DNA/St', 'kT']
+
+As we can see from the model equation, the model is given as distance as a function
+of force.
+
+
+Model composition and inversion
+-------------------------------
+
+In practice, we would typically want to fit force as a function of distance however. For this
+we have the inverted Odijk model::
+
+
+    odijk_inverted = pylake.inverted_odijk("DNA")
+
+
+We can have a quick look at what this model looks like and which parameters are in there::
+
+    >>> pylake.inverted_odijk("DNA")
+
+    Model equation:
+
+    f(d) = argmin[f](norm(DNA.Lc * (1 - (1/2)*sqrt(kT/(f*DNA.Lp)) + f/DNA.St)-(d - DNA.d_offset))) + DNA.f_offset
+
+    Parameter defaults:
+
+    Name      Value  Unit      Fitted      Lower bound    Upper bound
+    ------  -------  --------  --------  -------------  -------------
+    DNA/Lp    40     [nm]      True                  0            inf
+    DNA/Lc    16     [micron]  True                  0            inf
+    DNA/St  1500     [pN]      True                  0            inf
+    kT         4.11  [pN*nm]   False                 0              8
+
+This seems to be what we want. Force as a function of distance. Now let's say we have acquired data,
+but we notice that the template matching wasn't completely optimal. And that we had some force drift,
+while forgetting to reset the force back to zero. In this case, we can incorporate offsets in our
+model. We can introduce an offset in the independent parameter, by calling `subtract_independent_offset`
+on our model::
+
+    odijk_with_offset = pylake.inverted_odijk("DNA").subtract_independent_offset()
+
+If we also expect an offset in the dependent parameter, we can simply add an offset model to our
+model::
+
+    odijk_with_offsets = pylake.inverted_odijk("DNA").subtract_independent_offset() + pylake.force_offset("DNA")
```

Comment:
Regarding point 1 and 2:

This was a tough call for me too. I'm not a huge fan of the `.subtract_independent_offset()` either. I don't think there's a reason the addition operator couldn't be implemented for a distance offset, but it could generate some potentially unsafe things.

The way it's implemented now, the plus operator behaves like you'd expect from a plus operator i.e. `model_f() + model_g() + model_h()` => `f(x) + g(x) + h(x)`. Addition only being allowed for matching independent variables.

I'd find it very strange and unexpected if for one particular model it'd suddenly behave as
`model_f() + model_g() + model_h()` => `inv(inv(f)(y) + g(y)) + h(x)`.

Note that explicitly doing the inversions like `inv(inv(f)(y) + g(y)) + h(x)` is also quite slow, subtracting an offset is a special case that I treat with a specialized model that propagates sensitivities for that case specifically. Since it is by far the most common use case I feel that that specialization is justified since it is really about 20x-50x faster.

If you do want to implement it by operators you would have to be explicit now, i.e.:
```python
(lk.odijk("DNA") + lk.distance_offset("DNA")).invert() + force_offset("DNA")
```
(Note that I used the forward odijk now).

I figured it'd be much safer to be specific about only allowing addition of models that have matching independent and dependent variables, and not do implicit inversions. I figured keeping the API constrained in this way for now would prevent surprises that I may not yet be able to foresee.

In addition to that an offset is special in the sense that you can easily propagate it cheaply and it is the most common use case for this purpose.

Regarding point 3, at the moment no. Whereas at the moment, I'd say it makes sense to enforce the independent offset to be named the same, as it is really tied to that submodel whereas the offset is kind of outside of the other models in the construct. Let me illustrate that with an example:

```python
protein(d).subtract_independent_offset() + dna(d).subtract_independent_offset() + lk.force_offset("construct")`
Note how the offsets for the protein and dna are really subtracting the distance that goes into that model specifically, whereas the force offset applies to both.
```

Reply:
That all makes sense. Thanks! It would be useful to add some of that explanation to the tutorial. It might be too much to dump on readers upfront so consider restructuring the tutorial a little bit. Right now, it's divided into sections by stages of the fitting process and the entire tutorial runs through one fit, introducing more advanced concepts inline in every section. It might be nice to restructure it so that it runs through a very basic fit from start to finish and then does on to introduce more advanced concepts like composite models, inversion, and the related constraints.